docs/content/doc/installation/with-docker.en-us.md

```diff
@@ -307,77 +307,233 @@ To set required TOKEN and SECRET values, consider using Gitea's built-in [genera
 
-```bash
-volumes:
-  - /home/git/.ssh/:/data/git/.ssh
-```
+- To make the forwarding work, the SSH port of the container (22) needs to be mapped to the host port 2222 in `docker-compose.yml` . Since this port does not need to be exposed to the outside world, it can be mapped to the `localhost` of the host machine:
 
-Now a SSH key pair needs to be created on the host. This key pair will be used to authenticate the `git` user on the host to the container.
+  ```yaml
+  ports:
+    # [...]
+    - "127.0.0.1:2222:22"
+  ```
 
-```bash
-sudo -u git ssh-keygen -t rsa -b 4096 -C "Gitea Host Key"
-```
+- Next on the host create the `git` user which shares the same `UID`/ `GID` as the container values `USER_UID`/ `USER_GID`. These values can be set as environment variables in the `docker-compose.yml`:
+
+  ```yaml
+  environment:
+    - USER_UID=1000
+    - USER_GID=1000
+  ```
+
+- Mount `/home/git/.ssh` of the host into the container. Otherwise the SSH authentication cannot work inside the container.
+
+  ```yaml
+  volumes:
+    - /home/git/.ssh/:/data/git/.ssh
+  ```
+
+- Now a SSH key pair needs to be created on the host. This key pair will be used to authenticate the `git` user on the host to the container.
+
+  ```bash
+  sudo -u git ssh-keygen -t rsa -b 4096 -C "Gitea Host Key"
+  ```
+
+- Please note depending on the local version of ssh you may want to consider using `-t ecdsa` here.
+
+- `/home/git/.ssh/authorized_keys` on the host now needs to be modified. It needs to act in the same way as `authorized_keys` within the Gitea container. Therefore add the public key of the key you created above ("Gitea Host Key") to `~/git/.ssh/authorized_keys`.
+
+  ```bash
+  echo "$(cat /home/git/.ssh/id_rsa.pub)" >> /home/git/.ssh/authorized_keys
+  ```
+
+  Important: The pubkey from the `git` user needs to be added "as is" while all other pubkeys added via the Gitea web interface will be prefixed with `command="/usr [...]`.
+
+  `/home/git/.ssh/authorized_keys` should then look somewhat like
+
+  ```bash
+  # SSH pubkey from git user
+  ssh-rsa <Gitea Host Key>
+
+  # other keys from users
+  command="/usr/local/bin/gitea --config=/data/gitea/conf/app.ini serv key-1",no-port-forwarding,no-X11-forwarding,no-agent-forwarding,no-pty <user pubkey>
+  ```
+
+- The next step is to create the file that will issue the SSH forwarding from the host to the container. The name of this file depends on your version of Gitea:
+
+  - For Gitea v1.16.0+:
+
+    ```bash
+    cat <<"EOF" | sudo tee /usr/local/bin/gitea
+    #!/bin/sh
+    ssh -p 2222 -o StrictHostKeyChecking=no git@127.0.0.1 "SSH_ORIGINAL_COMMAND=\"\$SSH_ORIGINAL_COMMAND\" $0 $@"
+    EOF
+    chmod +x /usr/local/bin/gitea
+    ```
+
+  - For Gitea v1.15.x and earlier
+
+    ```bash
+    cat <<"EOF" | sudo tee /app/gitea/gitea
+    #!/bin/sh
+    ssh -p 2222 -o StrictHostKeyChecking=no git@127.0.0.1 "SSH_ORIGINAL_COMMAND=\"$SSH_ORIGINAL_COMMAND\" $0 $@"
```

Comment:
Should the `$` be escaped?

Reply:
The "EOF" heredoc means the $ gets put in the script as seen meaning we don't need to quote the $ and the $ expansion is occuring at the right place now. 

So the only question is whether we need to double quote to deal with whitespace. 

We don't need "$0" as the suggested file names don't include spaces but we could add.

The $@ shouldn't need double quoted as there are no places that a space should be in the argument. I'm not completely sure of the failure modes though as this is reporting as working without the double quoting as it stands and certainly works in my testing.

The problem with double quoting the "$@" is whether the this results in: "$1" "$2" "$3" ... or whether this results in "$1 $2 $3 ...". Normally "$@" should result in effectively the first expansion but it's already within a double quoted string so I don't know what will happen. 

I am suspicious though that double quoting might actually only work in some configs and not others but I dunno.

Have you been able to test and show that double quoting works here?

Reply:
For question 1 (the `$`), it's different for line 374 (`...=\"\$SSH_...`) and line 3874 (`...=\"$SSH_...`) , one is escaped while another isn't.

I think it should be `ssh {options} SSH_ORIGINAL_COMMAND="$SSH_ORIGINAL_COMMAND" "$0" "$@"`

https://superuser.com/a/270875

Reply:
Well, for question 2, it's so tricky to do correct quoting with bash (I just found this: https://unix.stackexchange.com/questions/518012 ) , so maybe we had better leave it as it was.

Reply:
ah sorry I think the \$ should be $ !